shell/src/test/java/tachyon/shell/TFsShellTest.java

```diff
@@ -658,4 +647,238 @@ public void duTest() throws IOException {
+    byte[] exp2 = BufferUtils.getIncreasingByteArray(20);
+    byte[] exp3 = BufferUtils.getIncreasingByteArray(30);
+    byte[] expect = new byte[exp1.length + exp2.length + exp3.length];
+    System.arraycopy(exp1, 0, expect, 0,                         exp1.length);
+    System.arraycopy(exp2, 0, expect, exp1.length,               exp2.length);
+    System.arraycopy(exp3, 0, expect, exp1.length + exp2.length, exp3.length);
+    
+    int ret = mFsShell.run(new String[] {"cat", "/testWildCards/*/foo*"});
+    Assert.assertEquals(ret, 0);
+    Assert.assertArrayEquals(mOutput.toByteArray(), expect);
+  }
+  
+  @Test
+  public void freeWildCardTest() throws IOException {
+    TFsShellUtilsTest.resetTachyonFileHierarchy(mTfs, WriteType.MUST_CACHE);
+    
+    TachyonConf tachyonConf = mLocalTachyonCluster.getMasterTachyonConf();
+    
+    int ret = mFsShell.run(new String[] {"free", "/testWild*/foo/*"});
+    CommonUtils.sleepMs(null, tachyonConf.getInt(Constants.WORKER_TO_MASTER_HEARTBEAT_INTERVAL_MS,
+        Constants.SECOND_MS) * 2 + 10);
+    Assert.assertEquals(0, ret);
+    Assert.assertFalse(mTfs.getFile(new TachyonURI("/testWildCards/foo/foobar1")).isInMemory());
+    Assert.assertFalse(mTfs.getFile(new TachyonURI("/testWildCards/foo/foobar2")).isInMemory());
+    Assert.assertTrue(mTfs.getFile(new TachyonURI("/testWildCards/bar/foobar3")).isInMemory());
+    Assert.assertTrue(mTfs.getFile(new TachyonURI("/testWildCards/foobar4")).isInMemory());
+    
+    ret = mFsShell.run(new String[] {"free", "/testWild*/*/"});
+    CommonUtils.sleepMs(null, tachyonConf.getInt(Constants.WORKER_TO_MASTER_HEARTBEAT_INTERVAL_MS,
+        Constants.SECOND_MS) * 2 + 10);
+    Assert.assertEquals(0, ret);
+    Assert.assertFalse(mTfs.getFile(new TachyonURI("/testWildCards/bar/foobar3")).isInMemory());
+    Assert.assertFalse(mTfs.getFile(new TachyonURI("/testWildCards/foobar4")).isInMemory());
+  }
+  
+  @Test
+  public void lsWildcardTest() throws IOException {
+    TFsShellUtilsTest.resetTachyonFileHierarchy(mTfs, WriteType.MUST_CACHE);
+    
+    TachyonURI expf1 = new TachyonURI("/testWildCards/foo/foobar1");
+    TachyonURI expf2 = new TachyonURI("/testWildCards/foo/foobar2");
+    TachyonURI expf3 = new TachyonURI("/testWildCards/bar/foobar3");
+    
+    String expected = "";
+    String format = "%-10s%-25s%-15s%-5s\n";
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(10),
+            TFsShell.convertMsToDate(mTfs.getFile(expf1).getCreationTimeMs()), "In Memory",
+            expf1.getPath());
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(20),
+            TFsShell.convertMsToDate(mTfs.getFile(expf2).getCreationTimeMs()), "In Memory",
+            expf2.getPath());
+    expected +=
+        String.format(format, FormatUtils.getSizeFromBytes(30),
+            TFsShell.convertMsToDate(mTfs.getFile(expf3).getCreationTimeMs()), "In Memory",
+            expf3.getPath());
```

Comment:
If we have an array of sizes and an array of paths, then these `exptected += ` can be put into a `for` loop.